daemon/src/model/cfd.rs

```diff
@@ -775,32 +824,60 @@ impl Cfd {
         Ok(self.event(event))
     }
 
-    pub fn roll_over(self, completed: RolloverCompleted) -> Result<Event> {
+    pub fn roll_over(self, completed: RolloverCompleted) -> Result<Option<Event>, RolloverError> {
         // TODO: Compare that the version that we started the rollover with is the same as the
         // version now. For that to work we should pass the version into the state machine
         // that will handle rollover and the pass it back in here for comparison.
         // Daniel says: I don't think the version check will help in addition to what we check in
         // can_rollover. What difference would it make? If we are completed rollover and are
         // still able to apply it we should, otherwise we might produce inconsistent state with the
         // other party.
-        if let (Err(e), RolloverCompleted::Succeeded { .. }) = (self.can_rollover(), &completed) {
-            bail!(e)
-        }
 
         let event = match completed {
+            // These are a bit weird but should go away with
+            // https://github.com/itchysats/itchysats/issues/958
+            // because we should never get here.
+            Completed::Failed {
+                error:
+                    error
+                    @
+                    (RolloverError::NoDlc
+                    | RolloverError::WasJustRolledOver
+                    | RolloverError::AlreadyRollingOver
+                    | RolloverError::NotRollingOver
+                    | RolloverError::Committed
+                    | RolloverError::Attested
+                    | RolloverError::Final
+                    | RolloverError::AlreadyExpired),
+                ..
+            } => {
+                tracing::debug!(order_id = %self.id, "Rollover was not started: {:#}", error);
+                return Ok(None);
+            }
```

Comment:
I'm not sure I understand the note, given that this is applied on top of the work that hopefully fixes #958, yet we still need to handle all these errors. What was your idea?

Reply:
The idea was that we would not emit a started event for cases like "was just rolled over" so really, there shouldn't be a case within the error of the completed handler.

A not completely thought through idea I had once is that with these actors operating on the CFD directly, creating one command per outcome might be better than wrapping it in an enum?

Also, the entirety of this explictely modelled error might be unnecessary because we no longer need to transport the error as a value across actors so the command can just fail and we log the error for cases where we don't produce an event here.

Hope that makes sense, sorry for the breviety.